fortuna/src/api/revelation.rs

```diff
@@ -59,14 +60,16 @@ pub async fn revelation(
         .get(&chain_id)
         .ok_or_else(|| RestError::InvalidChainId)?;
 
-    let maybe_request = state
-        .contract
-        .get_request(state.provider_address, sequence)
-        .await
-        .map_err(|_| RestError::TemporarilyUnavailable)?;
+    let maybe_request_fut = state.contract.get_request(state.provider_address, sequence);
+
+    let current_block_number_fut = state.contract.get_block_number();
+
+    let (maybe_request, current_block_number) =
+        try_join!(maybe_request_fut, current_block_number_fut)
+            .map_err(|_| RestError::TemporarilyUnavailable)?;
```

Comment:
Minor point but it might be nice to log which of these failed by logging the error just in case.